### PR TITLE
[UI] deep links to pipeline details page from start page

### PR DESCRIPTION
**Pull request by Bobgy**
/cc @jingzhang36 
/assign @SinaChavoshi 
/area frontend

Demo: https://youtu.be/JTTpzSzm_v4

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3086)

### CODE REVIEW

File: frontend/src/pages/GettingStarted.tsx [hunk 1]
```diff
@@ -86,7 +107,18 @@ cssRaw(`
 }
 `);
 
+const demoPipelines = [
+  '[Demo] Unified DSL - Taxi Tip Prediction Model Trainer',
```
Comment (rmgogogo) [inline frontend/src/pages/GettingStarted.tsx]: latest name is not "Unified DSL" but "TFX pipeline" or something, please check the latest codes.

Comment (Bobgy) [reply frontend/src/pages/GettingStarted.tsx]: Thanks! Updated referring to https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/config/sample_config.json

**Comment by numerology** [comment]
Thanks @Bobgy ! LGTM as a quick fix. As for a long-term solution, I'm wondering can we stop baking a predefined set of pipelines into our backend server? Instead, we might automatically download pipeline packages automatically when the backend server spin up. Then, IIUC, in resource manager we can get the actual link (which depends on the pipeline ID) and put it on the landing page.

Compile and publish the pipeline packages could be part of the releasing process. I think it may also help us to apply quick fix to samples as well.

WDYT? @IronPan @rmgogogo @jingzhang36

**Comment by Bobgy** [comment]
> Thanks @Bobgy ! LGTM as a quick fix. As for a long-term solution, I'm wondering can we stop baking a predefined set of pipelines into our backend server? Instead, we might automatically download pipeline packages automatically when the backend server spin up. Then, IIUC, in resource manager we can get the actual link (which depends on the pipeline ID) and put it on the landing page.
> 
> Compile and publish the pipeline packages could be part of the releasing process. I think it may also help us to apply quick fix to samples as well.
> 
> WDYT? @IronPan @rmgogogo @jingzhang36

Sounds like a good idea to me. Just confirmation, do we also make an extra api endpoint just to query sample ids?

**Comment by Bobgy** [comment]
I'd like to add three more changes:
1. let frontend import names from a json file
2. add a presubmit test check to verify names are synced with backend
3. add simple unit tests for this components.

will update next monday

**Comment by Bobgy** [comment]
@numerology Can we improve error message of this?
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/3096/kubeflow-pipeline-sample-test/1228980064243159041#1:build-log.txt%3A6347
https://github.com/kubeflow/pipelines/blob/646aa5ff645ee0c88c8a7b2d98ba7b57baf0f303/test/sample-test/utils.py#L40

**Comment by rmgogogo** [comment]
> I'd like to add three more changes:
> 
> 1. [x]  let frontend import names from a json file
> 2. [x]  add a presubmit test check to verify names are synced with backend
> 3. [x]  add simple unit tests for this components.
> 
> will update next monday
> 
> EDIT: these changes are all done. I think it's relatively safe now. WDUT?

OK. Thanks.

File: frontend/src/config/sample_config_from_backend.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+[
+  {
+    "name": "[Demo] XGBoost - Training with Confusion Matrix",
+    "description": "[source code](https://github.com/kubeflow/pipelines/blob/master/samples/core/xgboost_training_cm) [GCP Permission requirements](https://github.com/kubeflow/pipelines/blob/master/samples/core/xgboost_training_cm#requirements). A trainer that does end-to-end distributed training for XGBoost models.",
+    "file": "/samples/core/xgboost_training_cm/xgboost_training_cm.py.yaml"
+  },
+  {
+    "name": "[Demo] TFX - Taxi Tip Prediction Model Trainer",
```
Comment (rmgogogo) [inline frontend/src/config/sample_config_from_backend.json]: so how to sync this new file with the existing sample json file?
Sina just has another PR update the sample title.

Comment (rmgogogo) [reply frontend/src/config/sample_config_from_backend.json]: backend/src/apiserver/config/sample_config.json

**Comment by rmgogogo** [comment]
(if this one is hard to be continued, it's also find we just let the link points to the list, just want to save your bandwidth)

File: frontend/src/config/sample_config_from_backend.json (hunk 2 above)
Comment (Bobgy) [reply frontend/src/config/sample_config_from_backend.json]: Updated a little bit, I included error message that tells how to update.
https://github.com/kubeflow/pipelines/pull/3086/files#diff-afdea79a57383ab17c1cbd4013a3723aR13

**Comment by Bobgy** [comment]
> (if this one is hard to be continued, it's also find we just let the link points to the list, just want to save your bandwidth)

I think the solution is complete, can you check again?

**Comment by numerology** [comment]
> > Thanks @Bobgy ! LGTM as a quick fix. As for a long-term solution, I'm wondering can we stop baking a predefined set of pipelines into our backend server? Instead, we might automatically download pipeline packages automatically when the backend server spin up. Then, IIUC, in resource manager we can get the actual link (which depends on the pipeline ID) and put it on the landing page.
> > Compile and publish the pipeline packages could be part of the releasing process. I think it may also help us to apply quick fix to samples as well.
> > WDYT? @IronPan @rmgogogo @jingzhang36
> 
> Sounds like a good idea to me. Just confirmation, do we also make an extra api endpoint just to query sample ids?

IMHO that seems not a p0 requirement at this moment. Anyway no need to make such changes in this PR.

**Comment by Bobgy** [comment]
@numerology @renming, thank you for the suggestions. can one of you lgtm?

File: frontend/src/config/sample_config_from_backend.json (hunk 2 above)
Comment (rmgogogo) [reply frontend/src/config/sample_config_from_backend.json]: thanks, as long as test can catch up the diff and msg people to well sync them.

**Comment by rmgogogo** [comment]
/lgtm
Thanks. It's hard task.

**Pull request merged**